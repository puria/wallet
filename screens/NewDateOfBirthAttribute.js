import React from 'react';
import { connect } from 'react-redux';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import Button from '../application/components/Button/Button';
import styles from './styles';
import { saveDateOfBirth } from '../application/redux/actions/attributes';

const NewDateOfBirthAttribute = props => (
  <View style={styles.attributesManagementContainer}>
    { props.featureToggles['test'] && <Button name="addDateOfBirth" onPress={() => props.saveDateOfBirth(props.walletId)} /> }
  </View>
);

NewDateOfBirthAttribute.propTypes = {
  saveDateOfBirth: PropTypes.func.isRequired,
  walletId: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  walletId: state.wallet.id,
  featureToggles: state.featureToggles,
});

const mapDispatchToProps = dispatch => ({
  saveDateOfBirth: walletId => dispatch(saveDateOfBirth('01/01/1990', walletId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(NewDateOfBirthAttribute);
