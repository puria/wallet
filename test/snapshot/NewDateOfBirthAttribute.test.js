import React from 'react';
import configureStore from 'redux-mock-store';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import NewDateOfBirthAttribute from '../../screens/NewDateOfBirthAttribute';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

const initialState = {
  navigation: {},
  wallet: {
    id: '123',
  },
  featureToggles: {
    'test': true,
  }
};

it('renders NewDateOfBirthAttribute component', () => {
  const wrapper = shallow(
    <NewDateOfBirthAttribute />,
    { context: { store: mockStore(initialState) } },
  );
  expect(wrapper.dive()).toMatchSnapshot();
});
