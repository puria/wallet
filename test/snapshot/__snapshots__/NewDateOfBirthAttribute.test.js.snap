// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NewDateOfBirthAttribute component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewDateOfBirthAttribute
    featureToggles={
      Object {
        "test": true,
      }
    }
    saveDateOfBirth={[Function]}
    walletId="123"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Button
        enabled={true}
        name="addDateOfBirth"
        onPress={[Function]}
        style={Object {}}
      />,
      "style": Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFF",
        "margin": 16,
        "marginTop": 30,
        "paddingHorizontal": 20,
        "paddingVertical": 40,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "enabled": true,
        "name": "addDateOfBirth",
        "onPress": [Function],
        "style": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Button
          enabled={true}
          name="addDateOfBirth"
          onPress={[Function]}
          style={Object {}}
        />,
        "style": Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFF",
          "margin": 16,
          "marginTop": 30,
          "paddingHorizontal": 20,
          "paddingVertical": 40,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "enabled": true,
          "name": "addDateOfBirth",
          "onPress": [Function],
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
